### ISSUE TITLE
Bagudkompatibilitet i scripts

### ISSUE THREAD

**Issue by agnetemoos** [OS2borgerPC/os2borgerpc-admin-site#54]
I forhold til core scripts er jeg meget i tvivl om, hvor mange scripts i kategorien "Under udfasning" jeg skal medtage. Det er scripts der er indbygget i images. Der ligger en del sikkerhedsscripts i den kategori. 

Jeg kan se at de bliver kørt af en del  kommuner. Mange af dem laver usynlige ændringer. Jeg tænker derfor, at mange kommuner kører dem bare for en sikkerhed skyld, uden at det reelt er nødvendigt.

I admin systemet kan man se borger-pc versionen i konfigurationer for en maskine. Den er altså registreret.
![image](https://github.com/user-attachments/assets/d7d93dd1-932c-4aae-ac2e-ce47228be2eb)

1. Giver det mening at lave et cut-off og sige at maskiner med borgerpc-version under xx.yy er forældede. Man kunne markerer dem med et tegn eller baggrundsfarve i admin-systemet der viser at den er gal.
2. Hvilken version havner opgraderede BorgerPCer i? Har de en meningsfuld borgerpc-version?
3. I mange scripts står der bare at de er inbyggede i "nyere image-versioner". Det er ret uklart. Vi skal have en afdækning af hvad det betyder.

**Comment by agnetemoos** [comment]
Jeg kom i tanke om denne oversigt der faktisk beskriver problematikken. 
https://github.com/OS2borgerPC/os2borgerpc-scripts/blob/master/README.md

Måske for BorgerPC man skulle arbejde med et cut off ved version 5.0.0. Kører man version 3.x og 4.x er der virkelig mange scripts som kunerne skal huske at lægge på. Det tror jeg ikke alle har styr på.

@sunekochhansen @NeutraChikara 
Har I en kommentar til problemstillingen?

**Comment by agnetemoos** [comment]
> Hvilken version havner opgraderede BorgerPCer i? Har de en meningsfuld borgerpc-version?

Nu har jeg kigget grundigt på de 22.04 opgraderingsscripts der er lavet. 
De sætter ikke ikke `os2borgerpc_version`. De sætter kun `os_release` og `distribution`
Dvs. at os2borgerpc_version er fuldstændig meningsløs.

Nederst i det sidste opgraderingsscript sker der en hel masse. 
https://github.com/magenta-aps/os2borgerpc-scripts/blob/master/common/22_04_opgradering/os2borgerpc_lts_upgrade_in_place_step_4.sh

Jeg tror i praksis at det er indholdet af en hel række sikkerhedsscripts der bliver rulllet på. 

Måske kan vi antage at den kode er blevet kørt på alle opgraderede maskiner, så det er en slags baseline... Hmm.

**Comment by agnetemoos** [comment]
> Måske kan vi antage at den kode er blevet kørt på alle opgraderede maskiner, så det er en slags baseline... Hmm.

Nej, det kan vi ikke antage. Se her i scripets versionshistorik hvordan sikkerhedsfeatures er blevet drypvis tilføjet, sikkert mens scriptet har været i driftt. Dvs. at opdaterede BorgerPCer rundt omkring er i inkonsistent sikkerhedstilstand og ingen konfigurationsvariabel fortæller noget om tilstanden. 

https://github.com/OS2borgerPC/os2borgerpc-scripts/commits/master/common/22_04_opgradering/os2borgerpc_lts_upgrade_in_place_step_4.sh

**Comment by agnetemoos** [comment]
**Løsningsforslag**
Mit forslag er at vi samler en suite af sikkerhedsscripts som I kører på opgraderede BorgerPCere når I onboarder en kunde. I fortæller dem at det er for at opnå en fælles sikkerhedsmæssig baseline. Scriptene skal ikke være tilgængelige for kunderne blandt globale scripts.

For nye kunder som f. eks. Varde er det ikke nødvendigt. De installerer ud fra et image som I giver dem og dermed er de sikkerhedsmæssigt home safe.

**Comment by sunekochhansen** [comment]
Hej @agnetemoos,

1. Jeg synes det giver mening at kigge på ubuntu versionen. Forudsat at det er LTS versionen, så er maskiner under ubuntu 22.04 LTS ikke længere supporteret og dermed forældede. Jeg synes at scripts generelt bør være konstrueret på en sådan måde at de selv kan tage højde for ubuntu versionen, hvis der er nogle forskelle.
2. nej, det har de ikke så vidt jeg kan se. Der er to forskellige konfigurationer, "borgerpc_version", "_os_name" og "_os_release" der fortæller lidt.
Fra dokumentationen:

| Navn | Betydning | Kommentar |
|--------|--------|--------|
| os2borgerpc_version | Versionen på det OS2borgerPC-image, som blev installeret | Medmindre computeren geninstalleres, vil dette tal ikke ændre sig |
| _os_name | Navnet på operativsystemet |  |
| _os_release | Versionen af operativsystemet |  |

En opgradering vil ændre "\_os\_" konfigurationerne, men ikke borgerpc_version.
Man kunne overveje om borgerpc_version overhovedet giver mening, og om den på sigt bør fjernes.

3) Jeg har gennemgået den README.md  [nævnt i kommentar 2](https://github.com/OS2borgerPC/os2borgerpc-admin-site/issues/54#issuecomment-2523030534) og [OS2BorgerPC image 5.3.0](https://github.com/OS2borgerPC/os2borgerpc-scripts/blob/master/README.md#os2borgerpc-image-530) siger at alle anbefalede scripts er med i denne version.
Jeg har gennemgået alle scripts nævnt for denne og alle tidligere versioner, og holdt op mod hvilke scripts jeg kan se bliver kørt på ved byg af version 5.3.0.
(Jeg har fjernet dubletter.)

| Er refereret i byg | script |
|--------|--------|
| Ja | os2borgerpc/browser/firefox_global_policies.sh |
| Ja | os2borgerpc/sikkerhed/protect_terminal.sh |
| Ja | os2borgerpc/sikkerhed/dconf_disable_gnome_remote_desktop.sh |
| Ja | os2borgerpc/sikkerhed/dconf_run_prompt_toggle.sh |
| Ja | os2borgerpc/sikkerhed/polkit_policy_shutdown_suspend.sh |
| Ja | os2borgerpc/sikkerhed/adjust_settings_access.sh |
| Ja | os2borgerpc/udfases/dconf_disable_user_switching.sh |
| Ja | os2borgerpc/udfases/dconf_disable_lock_menu.sh |
| Nej | os2borgerpc/sikkerhed/prevent_crontab_persistence.sh |
| Nej | os2borgerpc/sikkerhed/desktop_toggle_writable.sh |
| Nej | os2borgerpc/sikkerhed/dconf_gnome_lock_menu_editing.sh |
| Nej | os2borgerpc/sikkerhed/network_manager_allow_user_changes.sh |

**Comment by agnetemoos** [comment]
Det er en meget fint overblik du der har lavet, Sune. Det må være nøglen til at komme videre.

Jeg kan bidrage med et script der er dukket op i Magentas repo for 4 mdr. siden.
https://github.com/magenta-aps/os2borgerpc-scripts/blob/master/common/system/collect_info_and_randomize_jobmanager.sh

Jeg tror det er et script deres kunder bliver bedt om at køre. Det må vi spørge Dennis om.

Det udfører mange forskellige ting. Men bl. a.  ser ud til at rette issues forbundet med `	os2borgerpc/sikkerhed/protect_terminal.sh`

**Comment by agnetemoos** [comment]
Her er beskrivelsen af 
https://github.com/magenta-aps/os2borgerpc-scripts/blob/master/common/system/collect_info_and_randomize_jobmanager.sh

> Dette script justerer tjek-ind-tidspunkterne, så de er fordelt på både tilfældige minuttal og tifældige sekundtal.
> Dette har til formål at sprede arbejdet for admin-sitet mere ligeligt ud, hvilket gerne skulle betyde, at admin-sitet og vores API bliver mere responsivt (hurtigere svartider).
> 
> Derudover indsamler scriptet information om fabrikant og model for computeren samt basal info om CPU og RAM.
> Denne information kan være relevant i forbindelse med fejlfinding.
> Den indsamlede information kan ses under "Konfigurationer" for den pågældende computer.
> Der er ikke tale om persondata.
> 
> Derudover foretager scriptet følgende justeringer:
> 
> - Hvis klienten på computeren er så gammel, at den ikke opdaterer sig selv, vil scriptet opdatere klienten.
> Da klienten er en central del af OS2borgerPC, er det vigtigt at sikre, at klienten holdes opdateret.
> 
> - Scriptet retter den fejl, som var relateret til den gamle udgave af "Sikkerhed - Juster adgang til terminalen", der bevirkede, at nogle genveje på superusers skrivebord ikke virkede korrekt.- 
> 
> - På OS2borgerPC med Ubuntu 22.04 vil scriptet blokere for GNOME Remote Desktop.
> Dette er præcis den samme ændring, som foretages af scriptet "Sikkerhed - Bloker for GNOME Remote Desktop (Fjernskrivebord)".
> 
> Alle de nævnte ændringer er alle indbygget i de nyeste images (OS2borgerPC 5.3.0 og OS2borgerPC Kiosk 2.1.0).
> VI anbefaler altid at installere ud fra vores nyeste images.
